compiler/parser/RdrHsSyn.hs

```diff
@@ -3158,14 +3180,22 @@ validateTemplate vtTemplateName tbd@TemplateBodyDecls{..}
     validateImplementsBlocks ::
          [Located ParsedImplementsDeclBlock]
       -> P [Located ValidImplementsDeclBlock]
-    validateImplementsBlocks = mapM $ traverse validateImplementsBlock
+    validateImplementsBlocks = mapM validateImplementsBlock
 
-    validateImplementsBlock :: ParsedImplementsDeclBlock -> P ValidImplementsDeclBlock
-    validateImplementsBlock (ParsedImplementsDeclBlock iface decls) = do
+    validateImplementsBlock :: Located ParsedImplementsDeclBlock -> P (Located ValidImplementsDeclBlock)
+    validateImplementsBlock (L loc (ParsedImplementsDeclBlock iface decls)) = do
       impls <- mapM validateMethodImpl (groupValDecls decls)
       checkMultipleDecls iface impls
       mapM_ (checkNumArgs iface) impls
-      pure (ValidImplementsDeclBlock iface impls)
+
+      let isViewImpl impl = rdrNameToString (vimdId impl) == "view"
+      let (viewImpls, nonViewImpls) = partition (isViewImpl . unLoc) impls
+      viewImpl <- case viewImpls of
+        [] -> pure $ Nothing
```

Comment:
similar question as above, do we not need to report an error here?

Reply:
Similar response to as above.

Reply:
Done adding TODO for this too